staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/convert.go

```diff
@@ -245,8 +245,13 @@ func newExtensions(s *apiextensions.JSONSchemaProps) (*Extensions, error) {
 		XEmbeddedResource: s.XEmbeddedResource,
 		XIntOrString:      s.XIntOrString,
 		XListMapKeys:      s.XListMapKeys,
-		XListType:         s.XListType,
-		XMapType:          s.XMapType,
+	}
+
+	if s.XListType != nil && *s.XListType != "atomic" {
+		ret.XListType = *s.XListType
```

Comment:
hmm... I don't like turning an explicit value into an implicit value. I'm also slightly surprised this doesn't break the round-trip test

Reply:
having two strings is a speciality of our CRD API. Structural intentionally abstracts that. But this one is slightly different to what we did before: before we were able to facefully represent the CRD schemas, but had another validation layer which restricted it (to structual schemas). This one is different because it normalizes.

Roundtrips probably do not break because we roundtrip valid Structural -> go-openapi/CRD schemas -> valid Structural.